packages/trpc/server/routers/viewer/teams/hasActiveTeamPlan.handler.ts

```diff
@@ -22,18 +23,18 @@ export const hasActiveTeamPlanHandler = async ({ ctx }: HasActiveTeamPlanOptions
     },
   });
 
-  if (!teams.length) return false;
+  if (!teams.length) return { isActive: false, isTrial: false };
 
   // check if user has at least on membership with an active plan
   for (const team of teams) {
     const teamBillingService = new InternalTeamBilling(team);
     const isPlanActive = await teamBillingService.checkIfTeamHasActivePlan();
     if (isPlanActive) {
-      return true;
+      return { isActive: true, isTrial: false };
     }
```

Comment:
I think we should get the trial status from the Stripe subscription. There might be other reasons why a subscription might not be active (ex. past due)

Reply:
Oh I see - good point. Hadn’t thought about that

Reply:
What happens with these cases now? Are the actions disabled without the badge as well?

Reply:
Could we add a requires an active team subscription badge?